### PR TITLE
feat(new-relic): add read-only New Relic alerts and metrics integration

### LINKED ISSUE #139: New Relic Integration

### PR DESCRIPTION
**Pull request by raildo**
Fixes #139

#### Describe the changes you have made in this PR -

Adds a read-only New Relic integration (NerdGraph): config/client/verifier/setup,
alert-incident and NRQL-metrics investigation tools, integration wiring
(registry/catalog/CLI/effective models), onboarding wizard, alert
routing/reporting/template, and docs.

This supersedes the previously-stacked PRs #4869, #4870, #4871, and #4872
(closed per request in #4870 and tracked in maintainer issue #4875), now
consolidated into a single PR with every Greptile/CodeQL finding from that
stack fixed, and the `specs/new-relic-integration/` planning docs dropped
(not needed in-tree):

- **Security:** redacted the real account email/id that had been committed
  to the spec docs (now removed entirely); restricted `base_url` to New
  Relic's documented US/EU hosts — the client attaches the API key header
  unconditionally, so an arbitrary host would have exfiltrated the credential.
- **Correctness:** fixed NRQL default-injection appending `SINCE` after an
  existing `LIMIT` clause (invalid clause order, rejected by NerdGraph);
  fixed the mutation-keyword guard flagging legitimate queries that merely
  contain a forbidden word inside a string literal (e.g. a `WHERE` filter);
  fixed truncation detection comparing against the caller's original limit
  instead of the vendor-capped limit actually executed, which hid real
  truncation on large requests.
- **Cleanup:** replaced inline-duplicated env var name literals with the
  canonical constants in `config/constants/new_relic.py`; removed an unused
  module-level constant flagged by CodeQL.

### Demo/Screenshot for feature changes and bug fixes -

https://github.com/user-attachments/assets/b8a740df-9383-479f-bb05-d81b5a947712

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

Read-only NerdGraph client (`run_nrql`, mandatory HTTP→errors→data checking
order) backing two tools: `query_new_relic_alerts` (pairs `NrAiIncident`
transition rows by `incidentId`, handles the vendor's inconsistent `event`
casing, HTML-entity-decodes `operator`/`title`) and `query_new_relic_metrics`
(arbitrary NRQL with default `SINCE`/`LIMIT` injection and a vendor-ceiling
clamp). Config validation restricts `base_url` to New Relic's documented
hosts so the API key header is never sent to an arbitrary endpoint. Wired
into the standard integration registry/catalog/CLI/effective-models path,
mirroring the existing Honeycomb/Datadog integrations already in the repo.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Please check **Allow edits from maintainers** if you would like assistance.

### CODE REVIEW

File: integrations/new_relic/client.py [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+        capped_limit = min(max(int(limit), 1), NEW_RELIC_NRQL_LIMIT_MAX)
+        clauses: list[str] = []
+        if priority:
+            clauses.append(f"priority = {_nrql_string_literal(priority)}")
+        if entity_name:
+            clauses.append(f"entity.name = {_nrql_string_literal(entity_name)}")
+        where_clause = f" WHERE {' AND '.join(clauses)}" if clauses else ""
+        nrql = (
+            f"{_INCIDENTS_SELECT}{where_clause} "
+            f"SINCE {int(since_minutes)} minutes ago LIMIT {capped_limit}"
```
[the remark this answers is not in the document]
Comment (raildo) [reply integrations/new_relic/client.py]: Fixed — the NRQL now sets `ORDER BY timestamp DESC` explicitly (verified against the live account: this is already NerdGraph's undocumented default, so this pins the invariant rather than changing behavior). A resolved incident's close row is always chronologically `>=` its open row, so newest-first ordering guarantees a LIMIT cutoff can only ever drop the (redundant) open row — never the close row, which alone already carries both `openTime` and `closeTime`. Added a regression test asserting the clause order in the sent NRQL.

File: docs/new_relic.mdx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+Service: new_relic
+Status: passed
+Detail: Connected to https://api.newrelic.com and validated account access
+```
+
+Inside the REPL: `/integrations verify new_relic` or `/verify new_relic`.
+
+## Alerts require an existing alert condition
+
+`query_new_relic_alerts` reads from New Relic's `NrAiIncident` event data, which only has rows once at least one alert condition has fired at least once. If your account has no alert policies and conditions configured yet, the tool will correctly return zero incidents — that's not a misconfiguration, there's simply nothing to read yet. Set up an alert condition in New Relic first if you want to see this tool return data.
```
[the remark this answers is not in the document]
Comment (raildo) [reply docs/new_relic.mdx]: Fixed — reworded to describe the observable behavior (no alert data until a condition has fired) without naming the internal function or event type, per the suggested rewrite.

**Comment by muddlebee** [comment]
@greptile review

**Comment by muddlebee** [comment]
Looks good overall, will take another deeper look

**Comment by muddlebee** [comment]
@greptile review

**Comment by muddlebee** [comment]
@raildo nice work :+1:

**Comment by muddlebee** [comment]
@greptile review
